src/main/java/io/supertokens/oauth/OAuth.java

```diff
@@ -90,6 +102,132 @@ public static OAuthAuthResponse getAuthorizationUrl(Main main, AppIdentifier app
         return new OAuthAuthResponse(redirectTo, cookies);
     }
 
+    //This more or less acts as a pass-through for the sdks, apart from camelCase <-> snake_case key transformation and setting a few default values
+    public static JsonObject registerOAuthClient(Main main, AppIdentifier appIdentifier, Storage storage, JsonObject paramsFromSdk)
+            throws TenantOrAppNotFoundException, InvalidConfigException, IOException, OAuthClientRegisterException,
+            NoSuchAlgorithmException, StorageQueryException {
+
+        OAuthStorage oauthStorage = StorageUtils.getOAuthStorage(storage);
+        String adminOAuthProviderServiceUrl = Config.getConfig(appIdentifier.getAsPublicTenantIdentifier(), main).getOAuthProviderAdminServiceUrl();
+
+        byte[] idBaseBytes = new byte[48];
+
+        while(true){
+            new SecureRandom().nextBytes(idBaseBytes);
+            String clientId = "supertokens_" + Utils.hashSHA256Base64UrlSafe(idBaseBytes);
```

Comment:
why do we have the supertokens_ prefix to client ids? I dont mind it, but just curious.

Reply:
It was @porcellus' request to do so.